plugins/custom-codecs/src/main/java/org/opensearch/index/codec/customcodecs/Lucene95CustomCodec.java

```diff
@@ -23,12 +23,37 @@
  * @opensearch.internal
  */
 public abstract class Lucene95CustomCodec extends FilterCodec {
+
+    /** Default compression level used for compression */
     public static final int DEFAULT_COMPRESSION_LEVEL = 3;
 
     /** Each mode represents a compression algorithm. */
     public enum Mode {
-        ZSTD,
-        ZSTD_NO_DICT
+        /**
+         * ZStandard mode with dictionary
+         */
+        ZSTD("ZSTD"),
+        /**
+         * ZStandard mode without dictionary
+         */
+        ZSTD_NO_DICT("ZSTDNODICT"),
+        /**
+         * Default ZStandard mode
+         */
+        ZSTD_DEFAULT("Lucene95CustomCodec");
```

Comment:
```suggestion
        ZSTD_DEPRECATED("Lucene95CustomCodec");
```

This is the one we need to fix the issue with codec naming, let's make it deprecated explictitly